server/controllers/thread.controller.js

```diff
@@ -22,4 +23,67 @@ function get(req, res, next) {
         .catch(next);
 }
 
-export default { get };
+function list(req, res, next) {
+    const queryObject = {
+        raw: true,
+        where: {},
+    };
+
+    if (req.query.limit) {
+        queryObject.limit = req.query.limit;
+    }
+
+    if (req.query.offset) {
+        queryObject.limit = req.query.limit;
+    }
+
+    const senders = [Sequelize.fn('ARRAY_AGG', Sequelize.fn('DISTINCT', Sequelize.col('from'))), 'senders'];
```

Comment:
What is the reason for basically building a raw query here? This somewhat defeats the purpose of using an ORM.

Reply:
Since a group by is needed for the summaries this seemed like the cleanest way to do aggregate functions.

This is the way sequelize suggests to do aggregates in the docs, Don't like it but it could be worse.  
```
Model.findAll({
  attributes: [[sequelize.fn('COUNT', sequelize.col('hats')), 'no_hats']]
});
```

Reply:
That's weird that they suggest using raw SQL, since the idea behind an ORM is to decouple the query language.